src/addons/model.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @fileoverview A general modelling class used by the Presenter's
+ * getModel function.
+ */
+
+import { set, Emitter } from '../microcosm'
+
+function isObservable(binding) {
+  return binding && typeof binding.subscribe === 'function'
+}
+
+function isCallable(binding) {
+  return binding && typeof binding.call === 'function'
+}
+
+function invoke(binding, repo, scope) {
+  if (isCallable(binding)) {
+    return binding.call(scope, repo.state, repo)
+  }
+
+  return binding
+}
+
+export default class Model extends Emitter {
+  /**
+   * @param {Microcosm} repo Track this Microcosm instance for updates
+   * @param {scope} scope Scope to invoke functional bindings
+   */
+  constructor(repo, scope) {
+    super()
+
+    this.repo = repo
+    this.scope = scope
+    this.bindings = {}
+    this.subscriptions = {}
+    this.value = {}
+
+    this.repo.on('change', this.compute, this)
+  }
+
+  /**
+   * Track an observable. Sending updates to a given key.
+   * @param {string} key
+   * @param {Observable} observable
+   */
+  track(key, observable) {
+    let last = this.subscriptions[key]
+    let next = observable.subscribe(value => this.set(key, value))
+
+    this.subscriptions[key] = next
+
+    if (last) {
+      last.unsubscribe()
+    }
+  }
+
+  /**
+   * @param {Object} bindings An set of key/value pairs for building a model
+   */
+  bind(bindings) {
+    this.bindings = {}
+
+    for (var key in bindings) {
+      let binding = bindings[key]
+
+      if (isObservable(binding)) {
+        this.track(key, binding)
+      } else {
+        this.bindings[key] = binding
+      }
+    }
+
+    this.compute()
+  }
+
+  /**
+   * Update a specific model key. Emits a change event
+   * @param {string} key
+   * @param {*} value
+   */
+  set(key, value) {
+    let next = set(this.value, key, value)
+
+    if (this.value !== next) {
+      this.value = next
+      this._emit('change', this.value)
```

Comment:
you have `this._emit('change', this.value)` here as well as in `compute`, if i'm reading this correctly, will that result in n+1 separate change emissions if there are n changes to `this.value`? And, will the last emission (in `compute`) be redundant since the the previous change event would have emitted the most up-to-date `this.value`?

those questions aside, my understanding of why you've done this is so that observables have a way to emit change events when something occurs there. am i on base there?